### PR TITLE
add support for voluntary exits

### PR DESCRIPTION
**Pull request by corverroos**
Adds support for voluntary exits.

Also use charon image from dockerhub.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -252,3 +252,20 @@ Keep checking in for updates, [here](https://github.com/ObolNetwork/charon/#supp
...
+9. How do I voluntary exit a validator?
+    - A voluntary exit is when a validator chooses to stop performing its duties, and exits the beacon chain permanently. To voluntarily exit, the validator must continue performing its validator duties until successfully exited to avoid penalties.
+    - Note: Quorum peers in the cluster need to perform this task to exit a validator.
+    - Create a new `exit_keys` folder next to `.charon/validator_keys`: `mkdir .charon/exit_keys`
+    - Copy the validator keys and passwords that you want to exit from the `validator_keys` folder to the `exit_keys` folder.
+      - E.g. to exit validator #4: `cp .charon/validator_keys/keystore/keystore-4* .charon/exit_keys/`
+      - Warning: all keys copied to the `exit_keys` folder will be exited, so be careful!
+    - Ensure the external network in `compose-volutary-exit.yml` is correct.
+      - Confirm the name of the exiting `charon-distributed-validator-node` docker network: `docker network ls`.
+      - If it isn't `charon-distributed-validator-node-dvnode`, then update `compose-volutary-exit.yml` accordingly.
```
Comment (OisinKyne) [inline README.md:265]:
```suggestion
      - If it isn't `charon-distributed-validator-node_dvnode`, then update `compose-volutary-exit.yml` accordingly.
```

**Comment by OisinKyne** [comment]
```
oisin@oisin-itx:~/code/charon-distributed-validator-node$ docker-compose -f compose-voluntary-exit.yml up
WARNING: Found orphan containers (charon-distributed-validator-node_charon_1, charon-distributed-validator-node_grafana_1, charon-distributed-validator-node_lighthouse_1, charon-distributed-validator-node_geth_1, 30701a4c91da_charon-distributed-validator-node_jaeger_1, charon-distributed-validator-node_teku_1, charon-distributed-validator-node_node-exporter_1, charon-distributed-validator-node_prometheus_1) for this project. If you removed or renamed this service in your compose file, you can run this command with the --remove-orphans flag to clean it up.
Creating charon-distributed-validator-node_teku-exit_1 ... done
Attaching to charon-distributed-validator-node_teku-exit_1
teku-exit_1  | Loading configuration...
teku-exit_1  | Fatal error in VoluntaryExit. Exiting
teku-exit_1  | tech.pegasys.teku.infrastructure.exceptions.InvalidConfigurationException: Failed to retrieve network spec from beacon node endpoint 'http://charon:3600/'.
teku-exit_1  | Details: Missing value for spec constant 'INACTIVITY_PENALTY_QUOTIENT_BELLATRIX'
teku-exit_1  |  at tech.pegasys.teku.cli.subcommand.RemoteSpecLoader.getSpec(RemoteSpecLoader.java:46)
teku-exit_1  |  at tech.pegasys.teku.cli.subcommand.VoluntaryExitCommand.initialise(VoluntaryExitCommand.java:232)
teku-exit_1  |  at tech.pegasys.teku.cli.subcommand.VoluntaryExitCommand.run(VoluntaryExitCommand.java:111)
teku-exit_1  |  at picocli.CommandLine.executeUserObject(CommandLine.java:1939)
teku-exit_1  |  at picocli.CommandLine.access$1300(CommandLine.java:145)
teku-exit_1  |  at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2358)
teku-exit_1  |  at picocli.CommandLine$RunLast.handle(CommandLine.java:2352)
teku-exit_1  |  at picocli.CommandLine$RunLast.handle(CommandLine.java:2314)
teku-exit_1  |  at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
teku-exit_1  |  at picocli.CommandLine$RunLast.execute(CommandLine.java:2316)
teku-exit_1  |  at picocli.CommandLine.execute(CommandLine.java:2078)
teku-exit_1  |  at tech.pegasys.teku.cli.BeaconNodeCommand.parse(BeaconNodeCommand.java:265)
teku-exit_1  |  at tech.pegasys.teku.Teku.start(Teku.java:63)
teku-exit_1  |  at tech.pegasys.teku.Teku.startFromCLIArgs(Teku.java:86)
teku-exit_1  |  at tech.pegasys.teku.Teku.main(Teku.java:40)
teku-exit_1  | Caused by: java.lang.IllegalArgumentException: Missing value for spec constant 'INACTIVITY_PENALTY_QUOTIENT_BELLATRIX'
teku-exit_1  |  at com.google.common.base.Preconditions.checkArgument(Preconditions.java:220)
teku-exit_1  |  at tech.pegasys.teku.spec.config.SpecConfigBuilder.validateNotNull(SpecConfigBuilder.java:251)
teku-exit_1  |  at tech.pegasys.teku.spec.config.SpecConfigBuilder.validateConstant(SpecConfigBuilder.java:237)
teku-exit_1  |  at tech.pegasys.teku.spec.config.SpecConfigBuilder$BellatrixBuilder.validate(SpecConfigBuilder.java:758)
teku-exit_1  |  at java.base/java.util.Optional.ifPresent(Unknown Source)
teku-exit_1  |  at tech.pegasys.teku.spec.config.SpecConfigBuilder.validate(SpecConfigBuilder.java:233)
teku-exit_1  |  at tech.pegasys.teku.spec.config.SpecConfigBuilder.build(SpecConfigBuilder.java:114)
teku-exit_1  |  at tech.pegasys.teku.spec.config.SpecConfigReader.build(SpecConfigReader.java:104)
teku-exit_1  |  at tech.pegasys.teku.spec.config.SpecConfigLoader.loadRemoteConfig(SpecConfigLoader.java:59)
teku-exit_1  |  at tech.pegasys.teku.cli.subcommand.RemoteSpecLoader.lambda$getSpec$0(RemoteSpecLoader.java:38)
teku-exit_1  |  at java.base/java.util.Optional.map(Unknown Source)
teku-exit_1  |  at tech.pegasys.teku.cli.subcommand.RemoteSpecLoader.getSpec(RemoteSpecLoader.java:38)
teku-exit_1  |  ... 14 more
charon-distributed-validator-node_teku-exit_1 exited with code 1
```

**Comment by OisinKyne** [comment]
```
12:46:56.711 DEBG parsigex   Aggregated threshold partial signatures {"duty": "32/exit", "peer": "outrageous-taxi"}
12:46:56.766 ERRO retry      Permanent failure calling bcast broadcast: eth2http: failed to submit voluntary exit: POST failed with status 400: {"code":400,"message":"BAD_REQUEST: Invalid object: gossip verification failed: ExitValidationError(Invalid(BadSignature))","stacktraces":[]} {"duty": "32/exit", "peer": "outrageous-taxi"}
12:47:00.000 DEBG sched      Slot ticked {"slot": 3650035}
```

**Comment by corverroos** [comment]
Bumped the hard-coded epoch to bellatrix-hard fork (for prater) to mitigate against: https://github.com/sigp/lighthouse/issues/3471

**Review by xenowits — APPROVED** [review]
LGTM!

**Review by dB2510 — APPROVED** [review]

**Pull request merged**